core/src/main/java/io/parsingdata/metal/data/Source.java

```diff
@@ -16,16 +16,11 @@
 
 package io.parsingdata.metal.data;
 
-import java.io.IOException;
 import java.math.BigInteger;
 
 public abstract class Source {
 
-    Slice slice(final long offset, final BigInteger length) {
-        return new Slice(this, offset, length);
-    }
```

Comment:
(For future reference:)

To me it seems intuitive to think that `Source`s create their own slices. Can you explain why it is not the responsibility of `Source` anymore to create `Slice`s?

Reply:
I would say that if a `Source` could be split into partial sources, then I agree completely. But in this case, we are _creating_ a `Slice`, so that is a more natural fit as a public factory method on `Slice` itself.

You could argue that given their relationship, perhaps `Slice` should be nested inside `Source`.